packages/ra-ui-materialui/src/layout/Title.tsx

```diff
@@ -11,6 +11,17 @@ export interface TitleProps {
     title?: string | ReactElement;
 }
 
+interface TitleTextProps {
+    text: string;
+    className: string;
+}
+
+function TitleText({ text, ...rest }: TitleTextProps) {
```

Comment:
This ensures both the UI title and document title are identical, and we only run the hook if title is a string.

Reply:
Nice. I think this component (and its type) should be exported, so it's better to put it in its own file.